compiler/rustc_mir/src/interpret/intrinsics/type_name.rs

```diff
@@ -134,7 +134,12 @@ impl<'tcx> Printer<'tcx> for AbsolutePathPrinter<'tcx> {
 
         self.path.push_str("::");
 
-        self.path.push_str(&disambiguated_data.data.as_symbol().as_str());
+        match disambiguated_data.data.get_name() {
+            DefPathDataName::Named(name) => self.path.write_str(&name.as_str()).unwrap(),
+            DefPathDataName::Anon { namespace } => {
+                write!(self.path, "{{{{{}}}}}", namespace).unwrap()
+            }
+        }
```

Comment:
This would benefit from `fmt::Display`, although, this seems to not want disambiguator.

Anyway, this one you can change from `{{closure}}` to `{closure}`, nothing should be relying on it (unlike legacy symbol names, which might be parsed) - presumably you want to do observable changes like that in a separate PR?